samples/metadata_query.py

```diff
@@ -0,0 +1,65 @@
+####
+# This script demonstrates how to use the metadata API to query information on a published data source
+#
+# To run the script, you must have installed Python 3.5 or later.
+####
+
+import argparse
+import logging
+from pprint import pprint
+
+import tableauserverclient as TSC
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Use the metadata API to get information on a published data source.')
+    # Common options; please keep those in sync across all samples
+    parser.add_argument('--server', '-s', required=True, help='server address')
+    parser.add_argument('--site', '-S', help='site name')
+    parser.add_argument('--token-name', '-n', required=True,
+                        help='name of the personal access token used to sign into the server')
+    parser.add_argument('--token-value', '-v', required=True,
+                        help='value of the personal access token used to sign into the server')
+    parser.add_argument('--logging-level', '-l', choices=['debug', 'info', 'error'], default='error',
+                        help='desired logging level (set to error by default)')
+    # Options specific to this sample
+    parser.add_argument('datasource_name')
```

Comment:
We should probably make this resilient to a `Null`, if you make the default value `None` does the query method insert a null for the variable?

(I forgot what I wrote :) )

Reply:
`datasource_name` is a required parameter. It does not have a default value. Invoking this script without a `datasource_name` prints

```
usage: metadata_query.py [-h] --server SERVER [--site SITE] --token-name TOKEN_NAME --token-value TOKEN_VALUE [--logging-level {debug,info,error}] datasource_name
metadata_query.py: error: the following arguments are required: datasource_name
```

Reply:
Sorry, I meant I think that it should just print all data sources if you don't supply a name. You don't have to if you really want to keep it to just 1 published data source, but I think it shows off GraphQL's flexibility to allow both.

Reply:
Now I got it...
Yes, querying all data sources works, too. Updated the sample to do so